kernelci/cli/docker.py

```diff
@@ -0,0 +1,203 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2022, 2023 Collabora Limited
+# Author: Guillaume Tucker <guillaume.tucker@collabora.com>
+# Author: Jeny Sadadia <jeny.sadadia@collabora.com>
+
+"""Tool to generate Dockerfiles from Jinja2 templates and build images"""
+
+
+import io
+import json
+import sys
+
+from jinja2 import Environment, FileSystemLoader
+import docker
+
+from .base import Command, sub_main
+
+
+class DockerCommand(Command):
+    """Base command class for docker sub-commands"""
+
+    args = Command.args + [
+        {
+            'name': 'image',
+            'help': "Docker image name, e.g. gcc-10:x86",
+        },
+    ]
+    opt_args = Command.opt_args + [
+        {
+            'name': '--arch',
+            'help': "CPU architecture, e.g. x86",
+        },
+        {
+            'name': '--fragment',
+            'action': 'append',
+            'help': "Extra fragments, e.g. kernelci",
+        },
+        {
+            'name': '--image-version',
+            'help': "Docker image version tag e.g. 20221011.0",
+        },
+        {
+            'name': '--prefix',
+            'default': 'kernelci/',
+            'help': "Docker image tag prefix",
+        },
+    ]
+
+    @classmethod
+    def _gen_image_name(cls, args):
+        base_name = args.prefix + args.image
+        tag_strings = ([args.arch] if args.arch else []) + (
+            args.fragment or [])
+        if args.image_version:
+            tag_strings.append(args.image_version)
+        tag_name = '-'.join(tag_strings)
+        name = ':'.join((base_name, tag_name)) if tag_name else base_name
+        return base_name, tag_name, name
+
+
+class DockerBuildGenerateCommand(DockerCommand):
+    """Base command class for docker generate and build sub-commands"""
+
+    TEMPLATE_PATH = 'config/docker'
```

Comment:
I think we'll also need to look in `/etc/kernelci/config/docker` as a standard installation path.  The `kernelci` Docker images already have all the config files installed there.  We might also add a command line argument to specify other paths for user-supplied templates but that's not strictly required right now.

Reply:
So how about changing this to a list?

```suggestion
    TEMPLATE_PATHS = ['config/docker', '/etc/kernelci/config/docker']
```

Reply:
Updated it.